component-photo-gallery/src/ImageList.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import ImageItem from './ImageItem.js';
+
+export default class ImageList extends Component {
+    state = { 
+        selected: null
+    }
+
+    render() {
+        const imageData = this.props.images;
+        const filteredData = imageData.filter(horned => {
+            if(!this.state.selected ) return true;
+                return horned.keyword === this.state.selected || horned.horns === Number(this.state.selected) || horned.title === this.state.selected;
+        })
+        const imageComponents = filteredData.map(image => <ImageItem horned={image} key={image.title}/>)
+           
+        const handleChange = e => {
+            this.setState({selected: e.target.value})
+        }
+
+        return (
+            <div id="list-main">
+                <div id="list-top">
+                    <select onChange={handleChange}>
+                        {this.props.images.map(image => <option key={image.title}>{image.keyword}</option>)}
+                    </select>
```

Comment:
Looks like there's no way to go back to showing all of your options. If you where to add this line you can get that functionality:
```
                    <select onChange={handleChange}>
                        <option value="" >All</option>
                        {this.props.images.map(image => <option key={image.title}>{image.keyword}</option>)}
                    </select>
```